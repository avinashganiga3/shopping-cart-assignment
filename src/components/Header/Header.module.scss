@use "../../styles/colors";
@use "../../styles/mixins";

.mainHeader {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.05),
    inset 0 -1px 0 rgba(0, 0, 0, 0.15);
  padding-top: 0.8rem;
}
div.header {
  display: flex;
  padding: 0;
  .logo {
    display: block;
    width: 100px;
    height: auto;
    margin-bottom: 0.8rem;
  }
  .navigations {
    display: flex;
    flex-direction: column;
    flex: 1;
    line-height: initial;
  }
  @include mixins.sm {
    padding: 0 1.6rem;
    .logo {
      width: 190px;
    }
  }
}
