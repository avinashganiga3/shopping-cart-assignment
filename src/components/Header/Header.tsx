import { Link } from "react-router-dom";
import AuthNavHeader from "../AuthNavHeader/AuthNavHeader";
import MainNav from "../MainNav";
import styles from "./Header.module.scss";

const Header = () => {
  return (
    <header className={styles.mainHeader}>
      <div className={`${styles.header} container`}>
        <Link to="/">
          <img
            src="/static/images/logo.png"
            srcSet="/static/images/logo.png, /static/images/logo_2x.png 2x"
            alt="Sabka Bazar Logo"
            className={styles.logo}
            width="150px"
            height="68px"
          />
        </Link>
        <div className={styles.navigations}>
          <AuthNavHeader />
          <MainNav />
        </div>
      </div>
    </header>
  );
};

export default Header;
