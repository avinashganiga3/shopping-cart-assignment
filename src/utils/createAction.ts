export type ActionWithPayload<T, P> = {
    type: T;
    payload: P;
  };
  
  export type Action<T> = {
    type: T;
  };
  
  export function createAction<T, P>(
    type: T,
    payload: P
  ): ActionWithPayload<T, P>;
  
  export function createAction<T>(
    type: T,
    payload: void
  ): Action<T>;
  
  export function createAction<T, P>(type: T, payload: P) {
    return { type, payload };
  }