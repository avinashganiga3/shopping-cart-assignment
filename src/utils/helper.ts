export const detectMob = () => {
  return window.innerWidth <= 830;
};
