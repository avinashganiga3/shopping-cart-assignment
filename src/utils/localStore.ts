import { CartItem } from "../context/CartContext";

class LocalStore {
  isLoggedIn: boolean = false;
  cartItems: CartItem[] = [];
  constructor() {
    if (this.getItem("isLoggedIn") === true) {
      this.isLoggedIn = true;
    }
    this.cartItems = this.getItem("cartItems") || [];
  }

  getItem(key: string) {
    const data = localStorage.getItem(key);
    if (data) {
      return JSON.parse(data);
    }
    return null;
  }
  setItem(key: string, value: string | object | boolean) {
    localStorage.setItem(key, JSON.stringify(value));
  }
  removeItem(key: string) {
    localStorage.removeItem(key);
  }
}

const ls = new LocalStore();
export default ls;
