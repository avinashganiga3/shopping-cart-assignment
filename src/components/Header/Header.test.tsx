import Header from "./index";
import { customRender } from "../../test/utils";

describe("Header component", () => {
  let wrapper = () => customRender(<Header />);
  it("should render Header component", () => {
    expect(wrapper()).toMatchSnapshot();
  });
});
