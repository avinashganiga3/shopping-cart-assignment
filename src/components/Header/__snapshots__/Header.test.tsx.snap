// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="mainHeader"
      >
        <div
          class="header container"
        >
          <a
            href="/"
          >
            <img
              alt="Sabka Bazar Logo"
              class="logo"
              height="86px"
              src="/static/images/logo.png"
              srcset="/static/images/logo.png, /static/images/logo_2x.png 2x"
              width="190px"
            />
          </a>
          <div
            class="navigations"
          >
            <nav
              class="nav"
            >
              <ul
                class="authHeader"
              >
                <li>
                  <a
                    href="/sign-in"
                  >
                    Sign In
                  </a>
                </li>
                <li>
                  <a
                    href="/register"
                  >
                    Register
                  </a>
                </li>
              </ul>
            </nav>
            <div
              class="mainNavContents"
            >
              <nav
                class="nav"
              >
                <ul>
                  <li>
                    <a
                      href="/"
                    >
                      Home
                    </a>
                  </li>
                  <li>
                    <a
                      href="/products"
                    >
                      Products
                    </a>
                  </li>
                </ul>
              </nav>
              <div
                class="cartBlock"
              >
                <button
                  class="cart"
                >
                  <svg
                    fill="#be2857"
                    width="20px"
                  >
                    cart.svg
                  </svg>
                  <div
                    class="cartCount"
                  >
                    0
                     items
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="mainHeader"
    >
      <div
        class="header container"
      >
        <a
          href="/"
        >
          <img
            alt="Sabka Bazar Logo"
            class="logo"
            height="86px"
            src="/static/images/logo.png"
            srcset="/static/images/logo.png, /static/images/logo_2x.png 2x"
            width="190px"
          />
        </a>
        <div
          class="navigations"
        >
          <nav
            class="nav"
          >
            <ul
              class="authHeader"
            >
              <li>
                <a
                  href="/sign-in"
                >
                  Sign In
                </a>
              </li>
              <li>
                <a
                  href="/register"
                >
                  Register
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="mainNavContents"
          >
            <nav
              class="nav"
            >
              <ul>
                <li>
                  <a
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li>
                  <a
                    href="/products"
                  >
                    Products
                  </a>
                </li>
              </ul>
            </nav>
            <div
              class="cartBlock"
            >
              <button
                class="cart"
              >
                <svg
                  fill="#be2857"
                  width="20px"
                >
                  cart.svg
                </svg>
                <div
                  class="cartCount"
                >
                  0
                   items
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
